build_tools/buildkite/dana/upload_benchmarks.py

```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/env python3
+# Lint as: python3
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Upload benchmark results to IREE Benchmark Dashboards.
+
+This script is meant to be used by Buildkite for automation.
+
+Example usage:
+  # Export necessary environment variables:
+  export IREE_DASHBOARD_URL=...
+  export IREE_DASHBOARD_API_TOKEN=...
+  # Then run the script:
+  python3 upload_benchmarks.py /path/to/benchmark/json/file
+"""
+
+import argparse
+import json
+import os
+import re
+import requests
+import subprocess
+import time
+
+IREE_GITHUB_COMMIT_URL_PREFIX = 'https://github.com/google/iree/commit'
+IREE_PROJECT_ID = 'IREE'
+
+# Number of total retries when post to dashboard.
+RETRY_COUNT = 5
+
+# A non-exhaustive list of models and their descriptions.
+# For models listed here we can provide a nicer description for them on
+# webpage.
+IREE_MODELS = {
+    'MobileNetV2 (imagenet) [TensorFlow]':
+        'MobileNet v2 from https://www.tensorflow.org/api_docs/python/tf/keras/applications/MobileNetV2',
+    'MobileNetV3Small (imagenet) [TensorFlow]':
+        'MobileNet v3 from https://www.tensorflow.org/api_docs/python/tf/keras/applications/MobileNetV3Small',
+    'MobileBertSquad [TensorFlow]':
+        'MobileBERT (SQuAD) from https://github.com/google-research/google-research/tree/master/mobilebert'
+}
+
+# A non-exhaustive list of phones and their descriptions.
+# For phones listed here we can provide a nicer description for them on
+# webpage.
+IREE_PHONES = {
+    'Pixel-4':
+        'Google Pixel 4 chipset: Snapdragon 855 (https://en.wikichip.org/wiki/qualcomm/snapdragon_800/855)',
+    'SM-G980F':
+        'Samsung Galaxy S20 chipset: Exynos 990 (https://en.wikichip.org/wiki/samsung/exynos/990)',
+}
+
+
+def execute(args,
+            cwd,
+            capture_output=False,
+            treat_io_as_text=True,
+            verbose=False,
+            **kwargs):
+  """Executes a command."""
+  if verbose:
+    print(f"+{' '.join(args)}  [from {cwd}]")
+  return subprocess.run(args,
+                        cwd=cwd,
+                        check=True,
+                        capture_output=capture_output,
+                        text=treat_io_as_text,
```

Comment:
We generally try to support Python 3.6+, which means the `text` argument isn't available (even more reason to alias it)

```suggestion
                        universal_newlines=treat_io_as_text,
```

See https://docs.python.org/3.6/library/subprocess.html#subprocess.run

Reply:
Used the common `get_output` for this now.